File: plotting/figure_3_7A.js
const fs = require('fs');
const path = require('path');
const {
  WIDTH,
  HEIGHT,
  COLORS,
  STROKES,
  FONTS,
  tag,
  svgOpen,
  svgClose,
  defs,
  titleText,
  captionText,
  legendSwatch,
  note,
  dottedLine,
  escapeXml,
} = require('./sharedStyle');

function roundedRect(x, y, w, h, r, options = {}) {
  return tag('rect', { x, y, width: w, height: h, rx: r, ry: r, fill: options.fill ?? 'none', stroke: options.stroke ?? COLORS.axis, 'stroke-width': options.strokeWidth ?? STROKES.secondary });
}

function arrow(x1, y1, x2, y2, color = COLORS.axis, width = 1.5, dotted = false) {
  return tag('line', { x1, y1, x2, y2, stroke: color, 'stroke-width': width, 'marker-end': 'url(#arrow)', 'stroke-dasharray': dotted ? '2 6' : undefined, 'stroke-linecap': 'round' });
}

function smallBullet(x, y, color = COLORS.axis) {
  return tag('circle', { cx: x, cy: y, r: 6, fill: color });
}

function similarityThumbnail(x, y, w, h) {
  const elements = [];
  // Frame
  elements.push(roundedRect(x, y, w, h, 10, { fill: 'none', stroke: COLORS.axis }));
  // Axes labels Ri (vertical) and ω/N (horizontal)
  elements.push(tag('text', { x: x + w / 2, y: y + h + 24, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.tick }, 'ω/N'));
  elements.push(tag('text', { x: x - 24, y: y + h / 2, transform: `rotate(-90 ${x - 24} ${y + h / 2})`, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.tick }, 'Ri'));
  // Soft regions
  elements.push(tag('rect', { x: x + 0.25 * w, y: y + 0.2 * h, width: 0.5 * w, height: 0.6 * h, fill: COLORS.conversion, 'fill-opacity': 0.18 })); // conversion belt
  elements.push(tag('ellipse', { cx: x + 0.8 * w, cy: y + 0.25 * h, rx: 0.16 * w, ry: 0.12 * h, fill: COLORS.shear, 'fill-opacity': 0.18 })); // shear-favored
  elements.push(tag('ellipse', { cx: x + 0.2 * w, cy: y + 0.75 * h, rx: 0.18 * w, ry: 0.12 * h, fill: COLORS.classical, 'fill-opacity': 0.12 })); // near-classical
  // Labels
  elements.push(tag('text', { x: x + 0.5 * w, y: y + 0.18 * h, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.note }, 'conversion belt'));
  elements.push(tag('text', { x: x + 0.8 * w, y: y + 0.25 * h - 10, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.note }, 'shear-favored'));
  elements.push(tag('text', { x: x + 0.2 * w, y: y + 0.78 * h, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.note }, 'near-classical'));
  return elements.join('');
}

function stackedGauge(x, y, w, h, shares) {
  // shares: { classical, conversion, interfacial, shear } sum to 1
  const order = [
    ['Classical', 'classical'],
    ['Conversion', 'conversion'],
    ['Interfacial', 'interfacial'],
    ['Shear-mediated', 'shear'],
  ];
  let curX = x;
  const elements = [roundedRect(x, y, w, h, 10, { fill: 'none', stroke: COLORS.axis })];
  for (const [label, key] of order) {
    const width = Math.max(0, Math.min(1, shares[key])) * w;
    elements.push(tag('rect', { x: curX, y, width, height: h, fill: COLORS[key] }));
    // Inline label
    elements.push(tag('text', { x: curX + width / 2, y: y + h / 2 + 6, 'text-anchor': 'middle', fill: '#000', 'font-family': FONTS.family, 'font-size': FONTS.axisLabel }, label));
    curX += width;
  }
  return elements.join('');
}

function renderFigureA(outDir) {
  const parts = [];
  parts.push(svgOpen());
  parts.push(defs());
  parts.push(tag('rect', { x: 0, y: 0, width: WIDTH, height: HEIGHT, fill: COLORS.background }));
  parts.push(titleText('Figure 3.7A — Composite attenuation concept'));

  // Layout constants
  const margin = { top: 110, right: 80, bottom: 90, left: 80 };

  // Inputs column (A)
  const inputsX = margin.left;
  const inputsY = margin.top + 40;
  const inputsW = 300;
  const inputsH = 520;
  parts.push(roundedRect(inputsX, inputsY, inputsW, inputsH, 14));
  parts.push(tag('text', { x: inputsX + 16, y: inputsY - 10, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel, 'font-weight': '700' }, 'Inputs'));

  const items = [
    'ω/N (frequency ratio)',
    'Ri (stability)',
    'kδ (interface thickness)',
    'CZ, At (contrast)',
    'θ (incidence), r (range)'
  ];
  let iy = inputsY + 40;
  for (const label of items) {
    parts.push(smallBullet(inputsX + 20, iy - 6));
    parts.push(tag('text', { x: inputsX + 40, y: iy, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel }, label));
    iy += 56;
  }

  // Similarity space cue (B)
  const simX = inputsX + inputsW + 60;
  const simY = margin.top;
  const simW = 520;
  const simH = 320;
  parts.push(similarityThumbnail(simX, simY, simW, simH));
  parts.push(tag('text', { x: simX + 8, y: simY - 10, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel, 'font-weight': '700' }, 'Similarity space cue'));

  // Mechanism weighting module (C)
  const modX = simX + 40;
  const modY = simY + simH + 30;
  const modW = 560;
  const modH = 260;
  parts.push(roundedRect(modX, modY, modW, modH, 16));
  parts.push(tag('text', { x: modX + 16, y: modY - 10, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel, 'font-weight': '700' }, 'Mechanism weighting (shares sum to 1)'));

  // Horizontal gauge (100% stacked)
  const gaugeX = modX + 30;
  const gaugeY = modY + 60;
  const gaugeW = modW - 60;
  const gaugeH = 70;
  parts.push(stackedGauge(gaugeX, gaugeY, gaugeW, gaugeH, { classical: 0.35, conversion: 0.4, interfacial: 0.1, shear: 0.15 }));
  parts.push(tag('text', { x: modX + modW - 80, y: gaugeY + gaugeH + 20, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.note }, '∑wi=1'));
  parts.push(tag('text', { x: modX + 30, y: gaugeY + gaugeH + 46, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': 16 }, escapeXml('shares guided by §3.2 similarity & §3.4–3.6 scalings')));

  // Input arrows to module
  const moduleCenterY = gaugeY + gaugeH / 2;
  for (let i = 0; i < items.length; i++) {
    const y = inputsY + 40 + i * 56 - 6;
    parts.push(arrow(inputsX + inputsW, y, modX, moduleCenterY));
  }

  // Combination node (D)
  const nodeX = modX + modW / 2;
  const nodeY = modY + modH + 70;
  // Sigma-like node (approximate with path)
  const sigmaPath = `M ${nodeX - 30} ${nodeY} C ${nodeX - 10} ${nodeY - 24}, ${nodeX + 10} ${nodeY - 24}, ${nodeX + 30} ${nodeY} C ${nodeX + 10} ${nodeY + 24}, ${nodeX - 10} ${nodeY + 24}, ${nodeX - 30} ${nodeY}`;
  parts.push(tag('path', { d: sigmaPath, stroke: COLORS.axis, 'stroke-width': STROKES.primary, fill: 'none', 'stroke-linecap': 'round', 'stroke-linejoin': 'round' }));
  parts.push(tag('text', { x: nodeX, y: nodeY + 48, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel }, 'Composite attenuation'));
  // Arrow from module to node
  parts.push(arrow(modX + modW / 2, modY + modH, nodeX, nodeY - 26));

  // Outputs column (E)
  const outX = modX + modW + 120;
  const outY = modY + 10;
  const outW = 360;
  const outH = 420;
  parts.push(roundedRect(outX, outY, outW, outH, 14));
  parts.push(tag('text', { x: outX + 16, y: outY - 10, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel, 'font-weight': '700' }, 'Predicted / measured outputs'));
  // Tiny icons
  const iconY0 = outY + 56;
  // TL(ω) curve icon
  parts.push(tag('path', { d: `M ${outX + 30} ${iconY0} C ${outX + 120} ${iconY0 - 24}, ${outX + 180} ${iconY0 + 18}, ${outX + 300} ${iconY0 - 10}`, stroke: COLORS.axis, 'stroke-width': STROKES.secondary, fill: 'none' }));
  parts.push(tag('text', { x: outX + 30, y: iconY0 - 14, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel }, 'TL(ω)'));
  // γ^2(ω)
  const iconY1 = iconY0 + 90;
  parts.push(tag('path', { d: `M ${outX + 30} ${iconY1} C ${outX + 120} ${iconY1 - 20}, ${outX + 190} ${iconY1 + 30}, ${outX + 300} ${iconY1 - 6}`, stroke: COLORS.axis, 'stroke-width': STROKES.secondary, fill: 'none' }));
  parts.push(tag('circle', { cx: outX + 180, cy: iconY1 + 10, r: 6, fill: COLORS.axis }));
  parts.push(tag('text', { x: outX + 30, y: iconY1 - 14, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel }, 'γ²(ω)'));
  // PSD width icon
  const iconY2 = iconY1 + 90;
  parts.push(tag('path', { d: `M ${outX + 40} ${iconY2 + 30} Q ${outX + 70} ${iconY2 - 10}, ${outX + 100} ${iconY2 + 30}`, stroke: COLORS.axis, 'stroke-width': STROKES.secondary, fill: 'none' }));
  parts.push(tag('path', { d: `M ${outX + 180} ${iconY2 + 30} Q ${outX + 230} ${iconY2 - 16}, ${outX + 280} ${iconY2 + 30}`, stroke: COLORS.axis, 'stroke-width': STROKES.secondary, fill: 'none' }));
  parts.push(tag('text', { x: outX + 30, y: iconY2 - 14, fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.axisLabel }, 'PSD(ω) width'));

  // Design implications note
  parts.push(arrow(outX + outW / 2, outY + outH, outX + outW / 2, outY + outH + 60));
  parts.push(tag('text', { x: outX + outW / 2, y: outY + outH + 86, 'text-anchor': 'middle', fill: COLORS.text, 'font-family': FONTS.family, 'font-size': FONTS.note }, 'Design implications: Choose ω, θ, Ri, kδ to target mechanisms; update priors with data.'));

  // Priors dotted back-arrow
  parts.push(tag('path', { d: `M ${modX + modW} ${modY + modH / 2} C ${modX + modW + 200} ${modY + modH / 2 - 100}, ${outX - 80} ${outY + outH + 40}, ${modX + modW / 2} ${modY + modH + 20}`,
    stroke: COLORS.gridDark, 'stroke-width': STROKES.dotted, 'stroke-dasharray': '2 6', fill: 'none', 'marker-end': 'url(#arrowColored)'}));
  parts.push(note(modX + modW + 10, modY + modH / 2 - 8, 'priors'));
  parts.push(tag('path', { d: `M ${modX + modW / 2} ${modY + modH + 20} C ${modX + modW / 2 - 100} ${modY + modH + 80}, ${outX} ${outY + outH + 30}, ${outX + outW} ${outY + outH + 10}`,
    stroke: COLORS.axis, 'stroke-width': STROKES.secondary, fill: 'none', 'marker-end': 'url(#arrow)'}));
  parts.push(note(outX + outW - 4, outY + outH + 4, 'posterior (with data)', 'end'));

  // Legend (bottom-right)
  const legX = WIDTH - 420;
  const legY = HEIGHT - 120;
  parts.push(legendSwatch(legX, legY, 'Classical', COLORS.classical));
  parts.push(legendSwatch(legX, legY + 26, 'Mode conversion', COLORS.conversion));
  parts.push(legendSwatch(legX, legY + 52, 'Interfacial', COLORS.interfacial));
  parts.push(legendSwatch(legX, legY + 78, 'Shear-mediated', COLORS.shear));

  // Caption
  parts.push(captionText('Figure 3.7A. Composite attenuation concept. Similarity variables feed a mechanism-weighting module that yields shares summing to unity; these combine into a composite prediction of attenuation and observables used to set priors and design experiments.'));

  parts.push(svgClose());
  const svg = parts.join('');
  const outPath = path.join(outDir, 'figure_3_7A.svg');
  fs.writeFileSync(outPath, svg, 'utf-8');
  return outPath;
}

if (require.main === module) {
  const outDir = process.argv[2] || path.join(__dirname, '..', 'figures');
  fs.mkdirSync(outDir, { recursive: true });
  const out = renderFigureA(outDir);
  console.log('Wrote', out);
}

module.exports = { renderFigureA };

File: figure_3_7a.py
import matplotlib.pyplot as plt
import matplotlib.patches as patches
from matplotlib.patches import FancyBboxPatch, ConnectionPatch
import numpy as np

# Set up the figure with exact specifications
fig, ax = plt.subplots(figsize=(20, 14))  # 2000x1400 px at 100 DPI
ax.set_xlim(0, 20)
ax.set_ylim(0, 14)
ax.axis('off')

# Color scheme
colors = {
    'classical': '#6B7280',
    'conversion': '#F4A261', 
    'interfacial': '#6A4C93',
    'shear': '#2A9D8F',
    'amber_band': '#F4A261',
    'dotted_line': '#9CA3AF'
}

# Font settings
title_font = {'fontsize': 28, 'fontweight': 'bold', 'fontfamily': 'sans-serif'}
label_font = {'fontsize': 22, 'fontfamily': 'sans-serif'}
tick_font = {'fontsize': 16, 'fontfamily': 'sans-serif'}
note_font = {'fontsize': 18, 'fontfamily': 'sans-serif'}
caption_font = {'fontsize': 15, 'style': 'italic', 'fontfamily': 'sans-serif'}

# Main title
ax.text(10, 13.5, 'Figure 3.7A — Composite attenuation concept', 
        ha='center', va='center', **title_font)

# (A) Inputs column (left, ~300 px wide)
inputs_box = FancyBboxPatch((0.5, 8), 3, 4, 
                           boxstyle="round,pad=0.1", 
                           facecolor='white', edgecolor='black', linewidth=2)
ax.add_patch(inputs_box)
ax.text(2, 11.8, 'Inputs', ha='center', va='center', **label_font)

# Input items with bullet points
input_items = [
    'ω/N (frequency ratio)',
    'Ri (stability)', 
    'kδ (interface optical thickness)',
    'CZ, At (impedance/Atwood)',
    'θ, r (incidence, range)'
]

y_positions = [11.2, 10.8, 10.4, 10.0, 9.6]
for i, (item, y) in enumerate(zip(input_items, y_positions)):
    # Bullet point
    ax.plot(0.8, y, 'o', markersize=4, color='black')
    # Text
    ax.text(1.0, y, item, va='center', **tick_font)

# Add "contrast" tag for CZ, At
ax.text(3.2, 10.0, 'contrast', ha='center', va='center', 
        fontsize=12, style='italic', color='gray')

# (B) Similarity space cue (center-top, ~520×320 px)
similarity_box = FancyBboxPatch((4.5, 9.5), 5.2, 3.2, 
                               boxstyle="round,pad=0.1",
                               facecolor='white', edgecolor='black', linewidth=1.5)
ax.add_patch(similarity_box)

# Similarity space axes
ax.plot([5, 9.5], [10.5, 10.5], 'k-', linewidth=1)  # x-axis
ax.plot([5, 5], [10.5, 12.5], 'k-', linewidth=1)    # y-axis

# Axis labels
ax.text(7.25, 10.2, 'ω/N', ha='center', va='center', **tick_font)
ax.text(4.7, 11.5, 'Ri', ha='center', va='center', **tick_font, rotation=90)

# Axis ticks
ax.plot([5, 5], [10.4, 10.6], 'k-', linewidth=1)  # y=0
ax.plot([9.5, 9.5], [10.4, 10.6], 'k-', linewidth=1)  # y=2
ax.plot([4.8, 5.2], [10.5, 10.5], 'k-', linewidth=1)  # x=0.2
ax.plot([4.8, 5.2], [12.5, 12.5], 'k-', linewidth=1)  # x=2.2

# Soft regions
conversion_region = patches.Ellipse((7.25, 11.5), 2.5, 1.2, 
                                   facecolor=colors['conversion'], alpha=0.3)
ax.add_patch(conversion_region)
ax.text(7.25, 11.5, 'conversion\nbelt', ha='center', va='center', 
        fontsize=12, style='italic')

shear_region = patches.Ellipse((6, 12.2), 1.5, 0.6, 
                              facecolor=colors['shear'], alpha=0.3)
ax.add_patch(shear_region)
ax.text(6, 12.2, 'shear-\nfavored', ha='center', va='center', 
        fontsize=10, style='italic')

classical_region = patches.Ellipse((8.5, 10.8), 1.5, 0.6, 
                                  facecolor=colors['classical'], alpha=0.3)
ax.add_patch(classical_region)
ax.text(8.5, 10.8, 'near-\nclassical', ha='center', va='center', 
        fontsize=10, style='italic')

# (C) Mechanism weighting module (center-middle, ~560×260 px)
weighting_box = FancyBboxPatch((4.2, 6.5), 5.6, 2.6, 
                              boxstyle="round,pad=0.15",
                              facecolor='white', edgecolor='black', linewidth=2)
ax.add_patch(weighting_box)
ax.text(7, 8.8, 'Mechanism weighting (shares sum to 1)', 
        ha='center', va='center', **label_font)

# Stacked horizontal bar
bar_y = 7.8
bar_width = 4.5
bar_height = 0.4

# Mechanism shares (example values)
shares = [0.35, 0.45, 0.05, 0.15]  # classical, conversion, interfacial, shear
mechanisms = ['Classical', 'Mode conversion', 'Interfacial', 'Shear-mediated']
mechanism_colors = [colors['classical'], colors['conversion'], 
                   colors['interfacial'], colors['shear']]

x_start = 5.2
x_pos = x_start
for i, (share, color, name) in enumerate(zip(shares, mechanism_colors, mechanisms)):
    width = share * bar_width
    rect = patches.Rectangle((x_pos, bar_y), width, bar_height, 
                           facecolor=color, edgecolor='black', linewidth=1)
    ax.add_patch(rect)
    
    # Label above the bar
    ax.text(x_pos + width/2, bar_y + 0.6, name, ha='center', va='center', 
            fontsize=14, fontweight='bold')
    
    x_pos += width

# Sum annotation
ax.text(9.8, 7.8, '∑wi = 1', ha='center', va='center', 
        fontsize=14, style='italic')

# Guidance text
ax.text(7, 7.2, 'shares guided by §3.2 similarity & §3.4–3.6 scalings', 
        ha='center', va='center', fontsize=12, style='italic')

# (D) Combination node (center-bottom)
sigma_x, sigma_y = 7, 5.5
sigma_circle = patches.Circle((sigma_x, sigma_y), 0.8, 
                            facecolor='white', edgecolor='black', linewidth=2)
ax.add_patch(sigma_circle)
ax.text(sigma_x, sigma_y, 'Σ', ha='center', va='center', 
        fontsize=24, fontweight='bold')
ax.text(sigma_x, sigma_y - 1.2, 'Composite\nattenuation', ha='center', va='center', 
        fontsize=14, fontweight='bold')

# Arrow from weighting to sigma
arrow1 = ConnectionPatch((7, 6.5), (sigma_x, sigma_y + 0.8), 
                        "data", "data", arrowstyle="->", 
                        shrinkA=5, shrinkB=5, mutation_scale=20, 
                        fc="black", linewidth=2)
ax.add_patch(arrow1)

# Priors back-arrow (dotted)
arrow2 = ConnectionPatch((sigma_x - 0.8, sigma_y), (6.2, 6.5), 
                        "data", "data", arrowstyle="->", 
                        shrinkA=5, shrinkB=5, mutation_scale=15, 
                        fc="black", linewidth=1.5, linestyle='--')
ax.add_patch(arrow2)
ax.text(6.5, 5.8, 'priors', ha='center', va='center', 
        fontsize=12, style='italic')

# Posterior arrow (solid)
arrow3 = ConnectionPatch((sigma_x + 0.8, sigma_y), (11.2, 5.5), 
                        "data", "data", arrowstyle="->", 
                        shrinkA=5, shrinkB=5, mutation_scale=20, 
                        fc="black", linewidth=2)
ax.add_patch(arrow3)
ax.text(9.5, 5.2, 'posterior (with data)', ha='center', va='center', 
        fontsize=12, style='italic')

# (E) Outputs column (right, ~360 px wide)
outputs_box = FancyBboxPatch((11.5, 6), 3.6, 4, 
                            boxstyle="round,pad=0.1", 
                            facecolor='white', edgecolor='black', linewidth=2)
ax.add_patch(outputs_box)
ax.text(13.3, 9.8, 'Predicted / measured outputs', ha='center', va='center', **label_font)

# Output icons
# TL(ω) curve
omega_vals = np.linspace(0.2, 2.2, 50)
tl_vals = 20 * np.exp(-(omega_vals - 1)**2) + 5
tl_x = 12 + (omega_vals - 0.2) / 2 * 1.5
tl_y = 9.2 + tl_vals / 100
ax.plot(tl_x, tl_y, 'k-', linewidth=2)
ax.text(13.3, 9.0, 'TL(ω)', ha='center', va='center', fontsize=12)

# Coherence γ²(ω) 
coh_vals = 1 - 0.8 * np.exp(-(omega_vals - 1)**2 / 0.1)
coh_x = 12 + (omega_vals - 0.2) / 2 * 1.5
coh_y = 8.2 + coh_vals * 0.3
ax.plot(coh_x, coh_y, 'k-', linewidth=2)
ax.text(13.3, 7.8, 'γ²(ω)', ha='center', va='center', fontsize=12)

# PSD(ω) width
psd_narrow = 0.1 * np.exp(-(omega_vals - 1)**2 / 0.05) + 0.05
psd_broad = 0.3 * np.exp(-(omega_vals - 1)**2 / 0.2) + 0.1
psd_x = 12 + (omega_vals - 0.2) / 2 * 1.5
ax.plot(psd_x, 7.2 + psd_narrow, 'k-', linewidth=2)
ax.plot(psd_x, 6.8 + psd_broad, 'k-', linewidth=2)
ax.text(13.3, 6.6, 'PSD(ω)', ha='center', va='center', fontsize=12)

# Design implications
ax.text(13.3, 5.8, 'Design implications:', ha='center', va='center', 
        fontsize=12, fontweight='bold')
ax.text(13.3, 5.4, 'Choose ω, θ, Ri, kδ to target\nmechanisms; update priors with data.', 
        ha='center', va='center', fontsize=10, style='italic')

# Legend (bottom-right)
legend_x = 15
legend_y = 3.5
legend_items = ['Classical', 'Mode conversion', 'Interfacial', 'Shear-mediated']
legend_colors = [colors['classical'], colors['conversion'], 
                colors['interfacial'], colors['shear']]

for i, (item, color) in enumerate(zip(legend_items, legend_colors)):
    y_pos = legend_y - i * 0.4
    rect = patches.Rectangle((legend_x, y_pos - 0.1), 0.3, 0.2, 
                           facecolor=color, edgecolor='black', linewidth=1)
    ax.add_patch(rect)
    ax.text(legend_x + 0.4, y_pos, item, va='center', fontsize=14)

# Caption
caption_text = ("Figure 3.7A. Composite attenuation concept. Similarity variables feed a "
                "mechanism-weighting module that yields shares summing to unity; these combine "
                "into a composite prediction of attenuation and observables used to set priors "
                "and design experiments.")
ax.text(10, 1.5, caption_text, ha='center', va='center', **caption_font, 
        bbox=dict(boxstyle="round,pad=0.5", facecolor='white', alpha=0.8))

# Input arrows to central module
for y in y_positions:
    arrow = ConnectionPatch((3.5, y), (4.2, 7.8), 
                          "data", "data", arrowstyle="->", 
                          shrinkA=5, shrinkB=5, mutation_scale=15, 
                          fc="black", linewidth=1.5)
    ax.add_patch(arrow)

plt.tight_layout()
plt.savefig('figure_3_7a.png', dpi=100, bbox_inches='tight', 
            facecolor='white', edgecolor='none')
plt.savefig('figure_3_7a.pdf', bbox_inches='tight', 
            facecolor='white', edgecolor='none')
plt.show()